pkg/Cpanel/Security/Advisor/Assessors/Imunify360.pm

```diff
@@ -142,15 +128,41 @@ sub _process_template {
     die "Template processing failed: $output";
 }
 
+sub create_purchase_link {
+    my ($self) = @_;
+
+    my $imunify    = Whostmgr::Imunify360->new();
+    my $installed  = $imunify->is_product_installed();
+    my $custom_url = $imunify->get_custom_url();
+    my $price      = $imunify->get_product_price();
+    my $cp_url     = $self->base_path('scripts12/purchase_imunify360_init');
+
+    if ($custom_url) {
+        return locale()->maketext( '[output,url,_1,Get Imunify360,_2,_3].', $custom_url, 'target', '_blank', );
```

Comment:
Just FYI: you can simplify by doing`locale->maketext` since `locale()` is defined at `BEGIN` time by the `use()` statement.

Reply:
On calls like this, I always intentionally include the parentheses because I've run into situations where Perl experienced ambiguity between the two different interpretations of a name like `locale`. Could be interpreted as:

1. The function call `locale()`, from which the return value is used as an instance on which `maketext` is called
2. The namespace `locale`, on which a static method `maketext` is called

Reply:
The ambiguity is usually only a problem when its defined locally after BEGIN, since its coming in during the use() its not likely to be ambiguous unless there actually is a namespace `locale` which would be indicative of a larger problem.

My rule of thumb for this sort of thing (i.e. chained methods) is if the parens are required there is probably something bad happening that I should understand and address

Reply:
I still find it kind of confusing to depend on Perl's preference to execute the thing as a function before treating it as a namespace. When you read the code, it looks exactly like a namespace, and only with knowledge of how Perl prioritizes the distinction between function calls and namespaces could you feel confident while reading the code that it's doing the correct thing. Obviously there is a trend at cPanel toward doing it the way you suggested, but I don't know if it's actually standardized that way.

To me, this also seems kind of like adding a review comment that a call to `time()` could be written simply as `time`. This is true, but it's such a trivial difference that it's hard to understand why it would be enforced as a rule.

Reply:
Sure, its not critical by any means. w/ `locale` and `logger` the intent was to not need parens because it helps make an otherwise noisy line slightly less noisy. same with any method chain really. `time()` usually doesn’t suffer from that noise. In the end its just a preference thing really `¯\_(ツ)_/¯`